### PR TITLE
ath79: add support for MikroTik hAP ac lite (RB952ui-5a2nd)

### PR DESCRIPTION
**Pull request by dmascord**
ath79: add support for MikroTik hAP ac lite (RB952Ui-5ac2nD)

    The MikroTik hAP ac lite (product code RB952Ui-5ac2nD) is
    an indoor 2.4Ghz and 5GHz AP with a 2 dBi integrated antenna built around the
    Atheros QCA9531 SoC.

    Specifications:
     - SoC: Atheros QCA9531
     - RAM: 64 MB
     - Storage: 16 MB NOR - Winbond 25Q128FVSG
     - Wireless: Atheros QCA9531 (SoC) 802.11b/g/n 2x2
     - Wireless: Atheros QCA9887 802.11a/n/ac 2x2
     - Ethernet: Atheros AR934X switch, 5x 10/100 ports,
        10-28 V passive PoE in port 1, 500 mA PoE out on port 5
     - 8 user-controllable LEDs:
      · 1x power (green)
      · 1x user (green)
      · 4x LAN status (green)
      · 1x WAN status (green)
      · 1x PoE power status (red)

     See https://mikrotik.com/product/RB952Ui-5ac2nD for more details.

    Notes:
     The device was already supported in the ar71xx target.
     WAN LED is not lit, though the port is functional.
     Serial console enablement needs version 3.27 of routerboot

    Flashing:
     TFTP boot initramfs image and then perform sysupgrade. Follow common
     MikroTik procedure as in https://openwrt.org/toh/mikrotik/common.

    Signed-off-by: Damien Mascord <tusker@tusker.org>

### CODE REVIEW

File: target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -8,6 +8,15 @@ board=$(board_name)
...
 
 case "$board" in
+mikrotik,routerboard-952ui-5ac2nd)
+	ucidef_set_led_timer "user" "USER" "mikrotik:green:user" "1000" "1000"
+	ucidef_set_led_netdev "wan" "WAN" "mikrotik:green:wan" "eth1"
+	ucidef_set_led_switch "lan2" "LAN2" "mikrotik:green:lan2" "switch0" "0x10"
+	ucidef_set_led_switch "lan3" "LAN3" "mikrotik:green:lan3" "switch0" "0x08"
+	ucidef_set_led_switch "lan4" "LAN4" "mikrotik:green:lan4" "switch0" "0x04"
+	ucidef_set_led_switch "lan5" "LAN5" "mikrotik:green:lan5" "switch0" "0x02"
+	ucidef_set_led_wlan "wlan" "WLAN" "mikrotik:green:wlan" "phy0tpt"
```
Comment (adschm) [inline target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds:18]: wifi led should be implemented via DT trigger.

File: target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds [hunk 2]
```diff
@@ -8,6 +8,15 @@ board=$(board_name)
 boardname="${board##*,}"
 
 case "$board" in
+mikrotik,routerboard-952ui-5ac2nd)
+	ucidef_set_led_timer "user" "USER" "mikrotik:green:user" "1000" "1000"
```
Comment (adschm) [inline target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds:12]: I'm not a fan of ucidef_set_led_timer, and it's almost never used nowadays, as "blinking" indicates boot or similar.
I'd just drop this line.

File: target/linux/ath79/mikrotik/base-files/etc/board.d/02_network [hunk 3]
```diff
@@ -22,6 +22,11 @@ ath79_setup_interfaces()
 	mikrotik,routerboard-wap-g-5hact2hnd)
 		ucidef_set_interface_lan "eth0"
 		;;
+	mikrotik,routerboard-952ui-5ac2nd)
+		ucidef_set_interfaces_lan_wan "eth0" "eth1"
```
Comment (adschm) [inline target/linux/ath79/mikrotik/base-files/etc/board.d/02_network:26]: ucidef_set_interface_wan "eth1"
is enough, lan is set via switch statement below.

**Comment by adschm** [comment]
Please have a look at the formal requirements for your commit title/message:

https://openwrt.org/submitting-patches

**Comment by dmascord** [comment]
Updated based on provided comments, thanks Adrian

File: target/linux/ath79/image/mikrotik.mk [hunk 4]
```diff
@@ -27,6 +27,16 @@ define Device/mikrotik_routerboard-922uags-5hpacd
 endef
 TARGET_DEVICES += mikrotik_routerboard-922uags-5hpacd
 
+define Device/mikrotik_routerboard-952ui-5ac2nd
+  $(Device/mikrotik_nor)
+  SOC := qca9533
+  DEVICE_MODEL := RouterBOARD 952Ui-5ac2nD (hAP ac lite)
+  DEVICE_PACKAGES += kmod-ath9k kmod-ath10k-ct-smallbuffers ath10k-firmware-qca9887-ct
```
Comment (adschm) [inline target/linux/ath79/image/mikrotik.mk:34]: kmod-ath9k can be dropped, it's selected via DEFAULT_PACKAGES

**Comment by dmascord** [comment]
Squashed into one commit

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+				#size-cells = <1>;
+
+				partition@0 {
+					label = "bootloader1";
+					reg = <0x0 0x0>;
+					read-only;
+				};
+
+				hard_config: hard_config {
+					#read-only;
```
Comment (f00b4r0) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:109]: Remove hash sign and alias (unused)

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+			gpios = <&gpio_ext 3 GPIO_ACTIVE_LOW>;
+		};
+		
+		lan5 {
+			label = "mikrotik:green:lan5";
+			gpios = <&gpio_ext 4 GPIO_ACTIVE_LOW>;
+		};
+		
+		wlan {
+			label = "mikrotik:green:wlan";
```
Comment (f00b4r0) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:52]: In ar71xx source this is marked as being blue, but I believe the LED isn't populated on the PCB anyway?

**Review by f00b4r0 — CHANGES_REQUESTED** [review]
Looks mostly good. Don't know what is the current policy wrt 03_gpio_switches: should it be edited to add the PoE/USB power out?

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts (hunk 6 above)
Comment (dmascord) [reply target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:52]: Let me confirm with the physical device, but I think you might be right

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts (hunk 5 above)
Comment (dmascord) [reply target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts]: Done

**Comment by adschm** [comment]
> Don't know what is the current policy wrt 03_gpio_switches: should it be edited to add the PoE/USB power out?

@f00b4r0 I don't really understand the question.

If this is about which mechanism, the current policy is that the author essentially can choose between gpio-hog, gpio-export and 03_gpio_switches based on what he/she prefers (though committers might still enforce something different if there is a reason). From common use, usb power is typically implemented via gpio-export, and poe passthrough is typically implemented via 03_gpio_switches. For poe power on specific ports I don't know.

**Comment by f00b4r0** [comment]
@adschm thanks for the clarification, I was confused about the various options. I guess the current situation in this PR is fine then.

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+				};
+
+				partition@10000 {
+					label = "bootloader2";
+					reg = <0x10000 0x0>;
+					read-only;
+				};
+
+				soft_config {
+					label = "soft_config";
```
Comment (f00b4r0) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:124]: Label is unneeded here. I missed that.

**Review by f00b4r0 — CHANGES_REQUESTED** [review]

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts (hunk 6 above)
Comment (dmascord) [reply target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:52]: There is no visible wlan LED, so I'll remove it

**Comment by dmascord** [comment]
All comments have been addressed, but I am having an issue with sysupgrade -c not actually keeping the config, even though it says it is:

Saving config files...
etc/config/ubootenv
etc/config/network
etc/config/firewall
etc/config/system
etc/config/dropbear
etc/config/wireless
etc/config/dhcp
etc/dropbear/dropbear_rsa_host_key
etc/dropbear/dropbear_ed25519_host_key
Commencing upgrade. Closing all shell sessions.
Watchdog handover: fd=3
- watchdog -
killall: telnetd: no process killed
Sending TERM to remaining processes ... logd hostapd wpa_supplicant netifd ntpd ubusd urngd
Sending KILL to remaining processes ...
Switching to ramdisk...
Performing system upgrade...
Unlocking firmware ...

Writing from <stdin> to firmware ...
Appending jffs2 data from /tmp/sysupgrade.tgz to firmware..

Upgrade completed
Rebooting system...

Then the booted system has reverted to stock settings (ie, 192.168.1.1 etc), and all changes have been lost.

**Comment by f00b4r0** [comment]
That's because 4K erase is broken, see #3271

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 8]
```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:2]: drop dts-v1

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+	compatible = "mikrotik,routerboard-952ui-5ac2nd", "qca,qca9533";
+	model = "MikroTik hAP ac lite";
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&led_wan_pin>;
+
+		power {
+			label = "mikrotik:green:power";
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:16]: drop mikrotik: prefix for all LEDs, yielding e.g. "green:power" for this one etc.

File: target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds [hunk 10]
```diff
@@ -8,6 +8,13 @@ board=$(board_name)
 boardname="${board##*,}"
 
 case "$board" in
+mikrotik,routerboard-952ui-5ac2nd)
+	ucidef_set_led_netdev "lan1" "WAN" "mikrotik:green:lan1" "eth1"
+	ucidef_set_led_switch "lan2" "LAN2" "mikrotik:green:lan2" "switch0" "0x10"
+	ucidef_set_led_switch "lan3" "LAN3" "mikrotik:green:lan3" "switch0" "0x08"
+	ucidef_set_led_switch "lan4" "LAN4" "mikrotik:green:lan4" "switch0" "0x04"
+	ucidef_set_led_switch "lan5" "LAN5" "mikrotik:green:lan5" "switch0" "0x02"
```
Comment (adschm) [inline target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds:16]: Drop mikrotik: here as well.

**Comment by SvenRoederer** [comment]
@dmascord just seen in the commit message: 5GHz is mentioned with 16dBI antenna. This can't be correct, as this would require some highly directional antenna, which the board does not have.

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+	};
+};
+
+&spi {
+	status = "okay";
+	
+	pinctrl-names = "default";
+	pinctrl-0 = <&pmx_spi_cs1>;
+
+	num-cs = <2>;
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:76]: num-cs can probably be dropped.

**Comment by apcameron** [comment]
@dmascord Any updates on this?

**Comment by apcameron** [comment]
Has there been any further progress on this?

**Comment by dmascord** [comment]
Apologies for the lack of progress, have been swamped with other work.   Will go through the comments above and push a new version for review

**Comment by dmascord** [comment]
I have tested the newly rebaselined version, and the sysupgrade problem seems to have been solved now, so that's good news :)

**Comment by rusitschka** [comment]
I'm applying this PR to 21.02 branch and can flash the image with a 19.07 initramfs. Everything works fine so far. What didn't work two times already was upgrading from 21.02 to a new 21.02 build via LuCI. I had to go back to 19.07 initramfs and flash 21.02 from there. Next time I'll try with SSH sysupgrade and see if that logs anything.

**Comment by rusitschka** [comment]
Here's the log of sysupgrade via shell. Doesn't say much and failed again. Any idea?
```
root@openwrt-office:/tmp# sysupgrade -v openwrt-ath79-mikrotik-mikrotik_routerboard-952ui-5ac2nd-squashfs-sysupgrade.bin
Sun Mar 14 15:02:36 CET 2021 upgrade: Saving config files...
etc/config/dhcp
etc/config/dropbear
etc/config/firewall
etc/config/luci
etc/config/network
etc/config/rpcd
etc/config/sqm
etc/config/system
etc/config/ubootenv
etc/config/ucitrack
etc/config/uhttpd
etc/config/wireless
etc/dropbear/authorized_keys
etc/dropbear/dropbear_ed25519_host_key
etc/dropbear/dropbear_rsa_host_key
etc/group
etc/hosts
etc/inittab
etc/luci-uploads/.placeholder
etc/opkg/keys/0b26f36ae0f4106d
etc/opkg/keys/1035ac73cc4e59e3
etc/opkg/keys/2f8b0b98e08306bf
etc/opkg/keys/5151f69420c3f508
etc/opkg/keys/72a57f2191b211e0
etc/opkg/keys/792d9d9b39f180dc
etc/opkg/keys/9ef4694208102c43
etc/opkg/keys/b2d571e0880ff617
etc/opkg/keys/b5043e70f9a75cde
etc/opkg/keys/c10b9afab19ee428
etc/opkg/keys/dace9d4df16896bf
etc/opkg/keys/dd57e4faf96d7e93
etc/opkg/keys/dd6de0d06bbd3d85
etc/opkg/keys/f790ad300b706d0d
etc/opkg/keys/f94b9dd6febac963
etc/passwd
etc/profile
etc/rc.local
etc/shadow
etc/shells
etc/shinit
etc/sysctl.conf
Sun Mar 14 15:02:37 CET 2021 upgrade: Commencing upgrade. Closing all shell sessions.
Connection to openwrt-office.localnet closed by remote host.
Connection to openwrt-office.localnet closed.
```

**Comment by f00b4r0** [comment]
see https://github.com/openwrt/openwrt/pull/3348#issuecomment-692530450

**Comment by rusitschka** [comment]
I see. But that PR has merge conflicts. @f00b4r0 Do you know if there is currently a solution for this?

**Comment by apcameron** [comment]
@dmascord What are the chances of having this backports to 21.02 in the near future?

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 12]
```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca9533_mikrotik_routerboard-16m.dtsi"
+
+/ {
+	compatible = "mikrotik,routerboard-952ui-5ac2nd", "qca,qca9533";
+	model = "MikroTik hAP ac lite";
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:7]: Please use the same name for compatible and model.

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+		compatible = "gpio-export";
+
+		gpio_usb_power {
+			gpio-export,name = "power:usb";
+			gpio-export,output = <1>;
+			gpios = <&gpio_ext 5 GPIO_ACTIVE_LOW>;
+		};
+		
+		poe_24v_port1 {
+			gpio-export,name = "10-28v-poe:lan5";
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:62]: Could this name be improved? E.g. like the node name?

BTW the property and the node name seem to have different meanings ...

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+			gpios = <&gpio_ext 4 GPIO_ACTIVE_LOW>;
+		};
+
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_usb_power {
+			gpio-export,name = "power:usb";
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:56]: I'd prefer something without colon, e.g. power-usb or usb-power.

File: target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+			label = "green:lan5";
+			gpios = <&gpio_ext 4 GPIO_ACTIVE_LOW>;
+		};
+
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_usb_power {
```
Comment (adschm) [inline target/linux/ath79/dts/qca9533_mikrotik_routerboard-952ui-5ac2nd.dts:55]: Just usb_power or power_usb should be enough.

File: target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds (hunk 10 above)
Comment (adschm) [reply target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds:15]: Ping.

**Comment by adschm** [comment]
Note that probably no reviewer will look at this if it's WIP ...

**Comment by dmascord** [comment]
@apcameron - Apologies, will clean up this PR, with the caveat that it won't work properly with sysupgrade due to the 4K/64K erase block size issue

**Comment by apcameron** [comment]
@adschm as a Number of the Mikrotik devices has this issue  "sysupgrade due to the 4K/64K erase block size issue" are there any plans to address it?

**Comment by adschm** [comment]
@apcameron The problem is that upstream seems to ignore it, and from our side nobody of the reviewer seemed particularly interested in it. So, I don't think this will be addressed anytime soon.

**Comment by dmascord** [comment]
@apcameron what is your need for this to be merged?  Would an automated unofficial master build with this PR and the dynamic erase size be of any use?

**Comment by apcameron** [comment]
I have one of these devices and the folks from [AREDN](https://github.com/aredn/aredn) are interested in a number of Mikrotik devices that we use currently on the ar71xx branch get ported to the ATH79 branch. So we would be interested in a stable solution that would work for the Mikrotik PR's that are currently out there for the ATH79 Branch.

**Comment by f00b4r0** [comment]
@adschm I suspect upstream is not fully aware of the need for the proposed change. It's possible the commit message wasn't quite clear enough on that, I've sent feedback to john on that topic, he seemed receptive, although I haven't heard from him since then. I strongly advocate to keep #3271 open, if only has a reminder / explanation for people looking for answers.

The other issue is I suspect that the 64K/4K mismatch is going to affect many other devices besides mikrotik's, given a quick `git grep 4K_SECTORS_LIMIT` shows that the default (in `target/linux/config-5.*`) is 4MB, meaning that devices with 4K-sector enabled AND flash larger than 4MB _expect_ to see 64K EB on large mtd partitions, an expectation that will not be met with current kernels. I see potential matches for these conditions in layerscape, octeon, rockchip, tegra and zynq, from git grep.

In other words, I wouldn't be surprised if a lot more people complained that their devices do not retain their configs once 21.02 is actually released and people start migrating.

@dmascord Unofficial build wouldn't solve the above problem, would cause confusion and would also probably make it impossible to use official repo's kernel modules (since the kernel has to be patched by #3271 )

HTH

**Comment by dmascord** [comment]
With #4379 this now is functional for sysupgrade, repeatably.  It does makes the sysupgrade process slow, so being very patient is important, so that you don't end up with a corrupt filesystem because you pulled the power too early.

**Comment by mkg20001** [comment]
Is there anything missing for merge? I'd need this for #4967

File: target/linux/ath79/image/mikrotik.mk [hunk 16]
```diff
@@ -37,6 +37,20 @@ define Device/mikrotik_routerboard-922uags-5hpacd
 endef
 TARGET_DEVICES += mikrotik_routerboard-922uags-5hpacd
 
+define Device/mikrotik_routerboard-952ui-5ac2nd
+  $(Device/mikrotik_nor)
+  SOC := qca9533
+  DEVICE_MODEL := RouterBOARD 952Ui-5ac2nD (hAP ac lite)
+  DEVICE_PACKAGES += kmod-ath10k-ct-smallbuffers ath10k-firmware-qca9887-ct
+  IMAGE_SIZE := 3712k
```
Comment (f00b4r0) [inline target/linux/ath79/image/mikrotik.mk:45]: Why? This looks wrong.

File: target/linux/ath79/mikrotik/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom [hunk 17]
```diff
@@ -29,6 +29,9 @@ case "$FIRMWARE" in
 	mikrotik,routerboard-wapr-2nd)
 		caldata_mikrotik_ath9k 0x1000 0x440 $(macaddr_add "$mac_base" 1)
 		;;
+	mikrotik,routerboard-952ui-5ac2nd)
+		caldata_mikrotik_ath9k 0x1000 0x440 $(macaddr_add "$mac_base" +6)
```
Comment (f00b4r0) [inline target/linux/ath79/mikrotik/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:33]: This looks unusual: does this match the sticker on the device?

Comment (mkg20001) [reply target/linux/ath79/mikrotik/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:33]: I have this in #4967 and the label mac matches

File: target/linux/ath79/image/mikrotik.mk [hunk 18]
```diff
@@ -37,6 +37,20 @@ define Device/mikrotik_routerboard-922uags-5hpacd
...
+  SYSUPGRADE_IMAGE_SIZE := 16256k
+  IMAGE/sysupgrade.bin := append-kernel | kernel2minor -s 1024 -e | \
+	pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | \
+	check-size $$$$(SYSUPGRADE_IMAGE_SIZE) | append-metadata
+  SUPPORTED_DEVICES += rb-952ui-5ac2nd
+endef
+TARGET_DEVICES += mikrotik_routerboard-952ui-5ac2nd
```
Comment (f00b4r0) [inline target/linux/ath79/image/mikrotik.mk:52]: This whole patch is wrong. See how it's done for the other NOR devices. You shouldn't need to redefine IMAGE/sysupgrade.bin or define SYSUPGRADE_IMAGE_SIZE

File: target/linux/ath79/mikrotik/base-files/etc/board.d/02_network [hunk 19]
```diff
@@ -22,6 +22,11 @@ ath79_setup_interfaces()
 	mikrotik,routerboard-wapr-2nd)
 		ucidef_set_interface_lan "eth0"
 		;;
+	mikrotik,routerboard-952ui-5ac2nd)
+		ucidef_set_interface_wan "eth1"
```
Comment (f00b4r0) [inline target/linux/ath79/mikrotik/base-files/etc/board.d/02_network:26]: Not defining lan (using `ucidef_set_interfaces_lan_wan`) will probably break MAC address assignment, as in https://github.com/openwrt/openwrt/pull/4967#issuecomment-1014381153

Can you check if this is the case?

**Comment by f00b4r0** [comment]
@mkg20001 do you plan to update this PR?

**Comment by mkg20001** [comment]
I don't have this particular device, I only have the other #4176. Not sure if I can.

**Comment by mkg20001** [comment]
Ah wait I meant to mention #4967

**Comment by apcameron** [comment]
@dmascord Are you planning to update this PR to resolve the Conflicts?

**Comment by f00b4r0** [comment]
@mkg20001 sorry I highlighted you in the wrong PR.

I would suggest you fix the problems I highlighted here in your own PR (mainly the `mikrotik.mk` bits which are wrong), remove this PR's code from your own PR and submit that. It'll speed things up.

I have the hAP-ac lite, if @dmascord does not update this PR I'll fix it and resubmit.
Thanks

File: target/linux/ath79/image/mikrotik.mk (hunk 18 above)
Comment (GektorUA) [reply target/linux/ath79/image/mikrotik.mk:52]: Can you show working patch? I want to compile latest OpenWrt for hAP AC Lite and test it.

**Comment by ynezz** [comment]
Superseded by #9825
